test/test_openapi.py

```diff
@@ -120,6 +121,67 @@ def route2():
                 pass
         self.assertEqual(str(cm.exception), 'Only specify path_spec once per route path')
 
+    def test_shared_path_spec(self):
+        app = AzulChaliceApp('testing', spec={'foo': 'bar'})
+        swagger_test_path_spec = {
+            'parameters': [
+                params.query('foo', schema.optional({'type': 'string'})),
+            ]
+        }
+
+        @app.route('/swagger-test-1', methods=['GET'], cors=True, path_spec=swagger_test_path_spec, method_spec={
```

Comment:
Duplicative.

Reply:
I removed some unnecessary content from the method spec so that this takes up less space, but for function definitions, I'm having a hard time thinking of a way to avoid at least _some_ duplication.

Reply:
Would a loop work?

![image](https://user-images.githubusercontent.com/5143256/81875792-f2f26a80-9535-11ea-94f9-a9b92e0a0705.png)

But note how both functions observe i at its final value. The decorator doesn't though which should work for you here.